command/agent/cache/lease_cache.go

```diff
@@ -89,6 +95,22 @@ type LeaseCacheConfig struct {
 	Logger      hclog.Logger
 }
 
+type inflightRequest struct {
+	// ch is closed by the request that ends up processing the set of
+	// parallel request
+	ch chan struct{}
+
+	// remaining is the number of remaining inflight request that needs to
+	// be processed before this object can be cleaned up
+	remaining atomic.Uint64
```

Comment:
Seems more natural to make this a sync.WaitGroup, what do you think?

Reply:
Oh nvm, I read too fast - you don't actually block on this.

Reply:
An earlier iteration of this used `sync.WaitGroup`, but that was harder to wrap my head around since the usage of remaining in here is for a non-blocking manner such that we know when we can clean up the `inflightCache` entry on L204. `Wait()` would block the caller, and we wouldn't know which request should call it to do the cleanup.